### PR TITLE
[Display_01] Built DisplayGpsInterfaceImplementation and Tests

### PR DESCRIPTION
**Pull request by DLidz**
Working on #67

### CODE REVIEW

**Review by stevecatherman — APPROVED** [review]
Yay!

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 1]
```diff
@@ -1,23 +1,67 @@
...
+import edu.oakland.production.display01.DisplayGpsReceiver;
 
 public class DisplayGpsInterfaceImplementation implements DisplayGpsInterface {
+  
+  public DisplayGpsReceiver reciever;
 
-  public boolean receiveGpsSignal(Satellite satelliteSignal) {
-    return true;
+  public DisplayGpsInterfaceImplementation(DisplayGpsReceiver reciever) {
+    this.reciever = reciever;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java:21]: Can you add a null checker? Any other class that is merged to main should be able to be used as an example. There is a test associated with this class as well.

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 2]
```diff
@@ -1,23 +1,67 @@
...
+  /** 
+   * Reports GPS Signal Loss.
+   *
+   * @param satelliteSignal GPS signal to report.
+   * @return Which GPS has Loss of Signal 
+   */
+  public SatelliteSignalCheckRequest reportGpsSignalLoss(Satellite satelliteSignal) {
+    System.out.println("Reporting loss for:");
+    SatelliteSignalCheckRequest satellite = reciever.measureSignal(satelliteSignal);
+    System.out.println(satellite.getSatelliteName());
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]: This is actually the next satellite to check, so this message should probably be something like:
```suggestion
    System.out.println("Attempting to " + satellite.getCheckType() + " " + satellite.getSatelliteName());
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 3]
```diff
@@ -1,23 +1,67 @@
...
-  public int recheckSignalStrength(Satellite satelliteSignal) {
-    return 0;
+  /** 
+   * Reports GPS Signal Loss.
+   *
+   * @param satelliteSignal GPS signal to report.
+   * @return Which GPS has Loss of Signal 
+   */
+  public SatelliteSignalCheckRequest reportGpsSignalLoss(Satellite satelliteSignal) {
+    System.out.println("Reporting loss for:");
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]: This should print the input satellite name:
```suggestion
    System.out.println("Reporting loss for " + satelliteSignal.getSatelliteName());
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 4]
```diff
@@ -1,23 +1,67 @@
...
+  /** 
+   * Rechecks GPS Signal Strength.
+   *
+   * @param satelliteSignal GPS signal to measure.
+   * @return Which GPS Rechecked 
+   */
+  public SatelliteSignalCheckRequest recheckSignalStrength(Satellite satelliteSignal) {
+    System.out.println("Rechecking signal strength");
+    SatelliteSignalCheckRequest satellite = reciever.measureSignal(satelliteSignal);
+    System.out.println(satellite.getSatelliteName());
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]: Same as above basically:
```suggestion
    System.out.println("Attempting to " + satellite.getCheckType() + " " + satellite.getSatelliteName());
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 5]
```diff
@@ -1,23 +1,67 @@
...
+  /** 
+   * Checks GPS Signal Strength.
+   *
+   * @param satelliteSignal GPS signal to measure.
+   * @return Which GPS Checked 
+   */
+  public SatelliteSignalCheckRequest checkSignalStrength(Satellite satelliteSignal) {
+    System.out.println("Checking signal strength");
+    SatelliteSignalCheckRequest satellite = reciever.measureSignal(satelliteSignal);
+    System.out.println(satellite.getSatelliteName());
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]: Same as above basically:
```suggestion
    System.out.println("Attempting to " + satellite.getCheckType() + " " + satellite.getSatelliteName());
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+import edu.oakland.production.display01.DisplayGpsInterfaceImplementation;
+import edu.oakland.production.display01.DisplayGpsReceiver;
+import edu.oakland.test.display01.DisplayGpsReceiverStub;
+import java.util.Random;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+
+@DisplayName("Display GPS Interface Unit Tests")
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java]: For standard log names, can we make this:
```suggestion
@DisplayName("DisplayGpsInterfaceImplementation Unit Tests")
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+public class DisplayGpsInterfaceImplementationTests {
+  @Test
+  @DisplayName("Verifying Signal In is Signal Out")
+  void signalInIsSignalOut() {
+    Satellite s = new Satellite("New Satellite", 7);
+    DisplayGpsInterfaceImplementation d;
+    d = new DisplayGpsInterfaceImplementation(new DisplayGpsReceiverStub());
+    assertEquals(d.receiveGpsSignal(s), "New Satellite");
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java]: Can we switch these? Expected always goes first according to the JUnit docs:
```suggestion
    assertEquals("New Satellite", d.receiveGpsSignal(s));
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  @Test
+  @DisplayName("Verifying Satellite Name in is Satellite Name Out")
+  void satelliteInIsSatelliteOutReportLoss() {
+    Satellite s = new Satellite("New Satellite", 7);
+    DisplayGpsInterfaceImplementation d;
+    d = new DisplayGpsInterfaceImplementation(new DisplayGpsReceiverStub());
+    SatelliteSignalCheckRequest ssrc = new SatelliteSignalCheckRequest("New Satellite", "Type 2");
+    assertEquals(d.reportGpsSignalLoss(s).getSatelliteName(), ssrc.getSatelliteName());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java]: Can we switch these? Expected always goes first according to the JUnit docs:
```suggestion
    assertEquals(ssrc.getSatelliteName(), d.reportGpsSignalLoss(s).getSatelliteName());
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  @Test
+  @DisplayName("Checking Satellite Signal")
+  void satelliteInIsSatelliteOutCheck() {
+    Satellite s = new Satellite("New Satellite", 7);
+    DisplayGpsInterfaceImplementation d;
+    d = new DisplayGpsInterfaceImplementation(new DisplayGpsReceiverStub());
+    SatelliteSignalCheckRequest ssrc = new SatelliteSignalCheckRequest("New Satellite", "Type 2");
+    assertEquals(d.checkSignalStrength(s).getSatelliteName(), ssrc.getSatelliteName());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java]: Can we switch these? Expected always goes first according to the JUnit docs:
```suggestion
    assertEquals(ssrc.getSatelliteName(), d.checkSignalStrength(s).getSatelliteName());
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  @Test
+  @DisplayName("Rechecking Satellite Signal")
+  void satelliteInIsSatelliteOutRecheck() {
+    Satellite s = new Satellite("New Satellite", 7);
+    DisplayGpsInterfaceImplementation d;
+    d = new DisplayGpsInterfaceImplementation(new DisplayGpsReceiverStub());
+    SatelliteSignalCheckRequest ssrc = new SatelliteSignalCheckRequest("New Satellite", "Type 2");
+    assertEquals(d.recheckSignalStrength(s).getSatelliteName(), ssrc.getSatelliteName());
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsInterfaceImplementationTests.java]: Can we switch these? Expected always goes first according to the JUnit docs:
```suggestion
    assertEquals(ssrc.getSatelliteName(), d.recheckSignalStrength(s).getSatelliteName());
```

File: src/test/java/edu/oakland/test/display01/DisplayGpsReceiverStub.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  /** 
+   * Measures GPS Signal Strength.
+   *
+   * @param satelliteSignal GPS signal to measure.
+   * @return Status of GPS signal measured.
+   */
+  public String measureGpsSignalStrength(Satellite satelliteSignal) {
+    Satellite s = satelliteSignal;
+    return s.getSatelliteName();
+
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display01/DisplayGpsReceiverStub.java]: We can streamline this to:
```suggestion
    return satelliteSignal.getSatelliteName();
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Couple changes, mostly around minor issues like print statements or ordering for JUnit. You should be able to accept all of the suggestions in a single batch from https://github.com/andrewdimmer/csi-3370-software-project/pull/117/files, then once you do so you should be ready to merge!

**Review by andrewdimmer — APPROVED** [review]
Looks good, you're set to merge!

**Pull request merged**